pkg/disk/partition_table.go

```diff
@@ -678,6 +689,62 @@ func (pt *PartitionTable) ensureLVM() error {
 	return nil
 }
 
+// ensureBtrfs will ensure that the root partition is on a btrfs subvolume, i.e. if
+// it currently is not, it will wrap it in one
+func (pt *PartitionTable) ensureBtrfs() error {
+
+	rootPath := entityPath(pt, "/")
+	if rootPath == nil {
+		panic("no root mountpoint for PartitionTable")
+	}
+
+	// we need a /boot partition to boot btrfs, ensure one exists
+	bootPath := entityPath(pt, "/boot")
+	if bootPath == nil {
+		_, err := pt.CreateMountpoint("/boot", 512*common.MiB)
```

Comment:
(super nitpick) I would avoid the extra `\n` here as the `if err != nil` is directly connected to the previous previous line. We could even considering just doing `if _, err .= ...; err != nil {` to make it slightly more compact